### PR TITLE
bumping enforce on key configs for `google_compute_security_policy` to GA

### PR DESCRIPTION
**Pull request by maxi-cit**
Fixes : https://github.com/hashicorp/terraform-provider-google/issues/14738

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:enhancement
compute: added `rule.rate_limit_options.enforce_on_key_configs` field to `google_compute_security_policy` resource (GA)
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/compute/resource_compute_security_policy.go.tmpl [hunk 1]
```diff
@@ -326,7 +326,6 @@ func ResourceComputeSecurityPolicy() *schema.Resource {
 										Description: `Rate limit key name applicable only for the following key types: HTTP_HEADER -- Name of the HTTP header whose value is taken as the key value. HTTP_COOKIE -- Name of the HTTP cookie whose value is taken as the key value.`,
 									},
 
-									{{ if ne $.TargetVersionName `ga` -}}
 									"enforce_on_key_configs": {
```
Comment (shuyama1) [inline mmv1/third_party/terraform/services/compute/resource_compute_security_policy.go.tmpl:329]: We'll probably also need to remove the version guard in the expander and flattener at the following places:
https://github.com/maxi-cit/magic-modules/blob/385c4194a932fab82cb02db8c2db294ae9a83e89/mmv1/third_party/terraform/services/compute/resource_compute_security_policy.go.tmpl#L1481

https://github.com/maxi-cit/magic-modules/blob/385c4194a932fab82cb02db8c2db294ae9a83e89/mmv1/third_party/terraform/services/compute/resource_compute_security_policy.go.tmpl#L1506

https://github.com/maxi-cit/magic-modules/blob/385c4194a932fab82cb02db8c2db294ae9a83e89/mmv1/third_party/terraform/services/compute/resource_compute_security_policy.go.tmpl#L1539

https://github.com/maxi-cit/magic-modules/blob/385c4194a932fab82cb02db8c2db294ae9a83e89/mmv1/third_party/terraform/services/compute/resource_compute_security_policy.go.tmpl#L1574

Comment (maxi-cit) [reply mmv1/third_party/terraform/services/compute/resource_compute_security_policy.go.tmpl:330]: Hey there, you re totally right. I forget about those.

**Comment by maxi-cit** [comment]
Hello @shuyama1, could you run test on GA please?

**Comment by shuyama1** [comment]
Triggered GA tests on [TC](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_MMUPSTREAMTESTS_GOOGLE_PACKAGE_COMPUTE/270259?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true)

**Comment by shuyama1** [comment]
Test status update: `TestAccComputeSecurityPolicy_EnforceOnKeyUpdates` still failed with same error.

**Comment by maxi-cit** [comment]
Hello @shuyama1, sry for the late reply. I found there was an additional conditional at the update func who was changing the behaviour between GA and Beta.

All tests should be fine now. Could you give it a check and rerun tests once more please?

**Comment by shuyama1** [comment]
Running GA tests on [TC](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_MMUPSTREAMTESTS_GOOGLE_PACKAGE_COMPUTE/276774?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true)

**Review by shuyama1 — APPROVED** [review]
GA tests passed. LGTM. Thank you!

**Pull request merged**